docx_editor/session.py

```
@@ -0,0 +1,430 @@
+        connection_file: Where to write the kernel connection file
+        timeout: Seconds to wait for the kernel to become ready
+
+    Returns:
+        PID of the kernel process.
+
+    Raises:
+        SessionError: If a session is already running or the kernel fails to start.
+        ImportError: If the [session] extra is not installed.
+    """
+    if is_session_running(connection_file):
+        raise SessionError(f"Session already running (connection file: {connection_file})")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)